app/controllers/pieces_controller.rb

```diff
@@ -12,10 +12,16 @@ def update
       flash[:danger] = "#{@piece.color.capitalize} #{@piece.type} move to X#{x}/Y#{y} was invalid"
     end
 
+    if @piece.game.en_passant_opportunity_active?
+      @piece.game.update_attribute(:allows_en_passant_capture?, true)
+    else
+      @piece.game.update_attribute(:allows_en_passant_capture?, false) if @piece.game.allows_en_passant_capture?
+    end
+
```

Comment:
Well, I guess you made me curious enough. I moved the logic in question to the piece move_to! method before the final return of true, and it seems to work fine. Not bad.